NewForce-Cohort-6/tabloidfullstack-cranberry-sauce
# Ticket #12 - Edit Category

# Description
This PR is for [Ticket #12](https://trello.com/c/jZHOt3o9/12-12-edit-a-category), which should allow users to select and edit a category from the list of categories.

> As an admin I would like to be able to modify a Category so that I can rephrase the name if I think of something more appropriate.
> 
> **Given** the user is viewing the Category list
> **When** they select the option to edit an Category
> **Then** the user should be directed to a form and given the ability to change the Category's name
> 
> **Given** the user is finished updating the Category information
> **When** they click the `Save` button
> **Then** the updated Category should be saved to the database
> **And** the user should be redirected to the new Category List page
> 
> **Given** the user has decided not to edit the Category
> **When** they click the `Cancel` button
> **Then** the user should be redirected back to the list page
> 
> > **NOTE:** For the time being it is acceptable to treat all users as `admin` users. There is a future story about enforcing user permissions.




# Testing Instructions
1. Commit or stash whatever you're currently working on
2. Git fetch --all
3. Git checkout ac-editCategory
4. If you don't see the most recent version of my files, git pull origin ac-editCategory
5.  If you need to create the Tabloid database, [use this file to do so](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/01_Tabloid_Create_DB.sql)
6. If you don't have the Tabloid seed data, [use this file to get it
](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/02_Tabloid_Seed_Data.sql)
7. Execute Tabloid from inside of Visual Studio
![Tabloid](https://user-images.githubusercontent.com/106983427/202474349-d57baf9b-323c-4406-b75a-d8f99a886603.png)
8.  Run NPM Start from inside of the Tabloid client folder
![client](https://user-images.githubusercontent.com/106983427/202473962-42d0b50f-1420-4b34-a522-464dc9fc11d5.png)
9. Login using an email that you know is there and works for you (example: foo@bar.com from seed data)
10. Once logged-in, use the navbar to navigate to category management ("/category")
11. Click the edit button located next to the name of each category
12. You should be redirected to a page ("/category/edit/id") containing a single input field, a save button, and a cancel button
13. Enter a new category name and click save to be redirected to the list of categories
14. You should see that the selected category name has changed to reflect your edits 

# Checklist:
